modules/distributor/distributor.go

```diff
@@ -526,20 +557,39 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 	return keys, traces, nil
 }
 
-func recordDiscaredSpans(err error, userID string, spanCount int) {
-	s := status.Convert(err)
-	if s == nil {
-		return
-	}
-	desc := s.Message()
+func countDiscaredSpans(responses []*tempopb.PushResponse, traces []*rebatchedTrace, repFactor int) (int, int) {
+	maxLiveDiscardedCount := 0
+	traceTooLargeDiscardedCount := 0
```

Comment:
Nit: you can used named return values, makes the method more explicit + you don't need to initialise these

```suggestion
func countDiscaredSpans(responses []*tempopb.PushResponse, traces []*rebatchedTrace, repFactor int) (maxLiveDiscardedCount int, traceTooLargeDiscardedCount int) {
```

The line gets a bit long though... 🤔